### PR TITLE
Add new `x-async-support` extension for generating async controllers

### PR DESCRIPTION
**Pull request by jeet23**
Recently a new capability to generate `CompletionStage` wrapped controllers was added in #304. However, it was using an `--http-controller-opts` option which does it for all or none controllers.

There are situations where clients require CompletionStage for only specific endpoints in a given controller. This PR adds a fine-grained control to specify which endpoint requires CompletionStage.

OpenAPI's [extensions](https://swagger.io/docs/specification/openapi-extensions/) are used to denote it. If you decorate the path of an endpoint with `x-async-support: true`, the controller generated will be wrapped with `CompletionStage`.

### CODE REVIEW

File: src/main/kotlin/com/cjbooms/fabrikt/generators/controller/SpringControllerInterfaceGenerator.kt [hunk 1]
```diff
@@ -78,7 +79,8 @@ class SpringControllerInterfaceGenerator(
             .addSpringFunAnnotation(op, verb, path.pathString)
             .addSuspendModifier()
 
-        val funcSpec = if (options.contains(ControllerCodeGenOptionType.COMPLETION_STAGE)) {
+        val asyncSupport = op.extensions.containsKey(EXTENSION_ASYNC_SUPPORT)
+        val funcSpec = if (options.contains(ControllerCodeGenOptionType.COMPLETION_STAGE) || asyncSupport) {
```
Comment (cjbooms) [inline src/main/kotlin/com/cjbooms/fabrikt/generators/controller/SpringControllerInterfaceGenerator.kt:83]: Should we allow asyncSupport to override default?
`x-async-support: false`

File: src/main/kotlin/com/cjbooms/fabrikt/generators/controller/SpringControllerInterfaceGenerator.kt [hunk 2]
```diff
@@ -78,7 +79,9 @@ class SpringControllerInterfaceGenerator(
             .addSpringFunAnnotation(op, verb, path.pathString)
             .addSuspendModifier()
 
-        val funcSpec = if (options.contains(ControllerCodeGenOptionType.COMPLETION_STAGE)) {
+        val asyncSupport =
```
Comment (cjbooms) [inline src/main/kotlin/com/cjbooms/fabrikt/generators/controller/SpringControllerInterfaceGenerator.kt:82]: Would rather see this changed to something like below:
```
val explicitAsyncSupport = op.extensions[EXTENSION_ASYNC_SUPPORT] as? Boolean
val asyncSupport = explicitAsyncSupport ?: options.contains(ControllerCodeGenOptionType.COMPLETION_STAGE)

val funcSpec = if (asyncSupport) {...

```

Comment (jeet23) [reply src/main/kotlin/com/cjbooms/fabrikt/generators/controller/SpringControllerInterfaceGenerator.kt]: Done

File: src/main/kotlin/com/cjbooms/fabrikt/generators/controller/SpringControllerInterfaceGenerator.kt (hunk 1 above)
Comment (jeet23) [reply src/main/kotlin/com/cjbooms/fabrikt/generators/controller/SpringControllerInterfaceGenerator.kt]: Added the support for handling both true and false.

**Pull request merged**